src/main/java/in/projecteka/consentmanager/link/link/Link.java

```diff
@@ -175,13 +178,11 @@ private String serializeLinkReferenceResultFromHIP(PatientLinkReferenceResult re
             String patientId,
             String hipId,
             UUID requestId) {
-        return linkServiceClient.confirmPatientLink(toLinkConfirmationRequest(patientLinkRequest, requestId), hipId)
-                .zipWith(Mono.delay(Duration.ofMillis(getExpectedFlowResponseDuration())))
-                .flatMap(tuple ->
-                        linkResults.get(requestId.toString())
-                                .switchIfEmpty(Mono.error(ClientError.gatewayTimeOut()))
-                                .flatMap(lcr -> deserializeConfirmationFromHIP(lcr)))
-                .switchIfEmpty(Mono.error(ClientError.networkServiceCallFailed()))
+        return scheduleThis(linkServiceClient.confirmPatientLink(toLinkConfirmationRequest(patientLinkRequest, requestId), hipId))
+                .timeout(Duration.ofMillis(getExpectedFlowResponseDuration()))
+                .responseFrom(discard -> Mono.defer(() -> linkResults.get(requestId.toString())))
+                .onErrorResume(DelayTimeoutException.class, discard -> Mono.error(ClientError.gatewayTimeOut()))
+                .flatMap(response -> tryTo(response, LinkConfirmationResult.class).map(Mono::just).orElse(Mono.empty()))
```

Comment:
you can remove unused method which was used for Deserialization and you can use `from` method in Serializer also and remove the method in this class.

Reply:
I didn't - thinking that Mounica might be using the same.